File: 2st/Riyeon/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int w=0, h=0;
        for(int[] s : sizes){
            if(s[0] < s[1]){
                int temp = s[0];
                s[0] = s[1];
                s[1] = temp;
            }
            if(w<s[0]) w = s[0];
            if(h<s[1]) h = s[1];
        }
        return w*h;
    }
}

File: 2st/Yeonwoo/최소직사각형.py
def solution(sizes):
    answer = 0
    temp = []
    temp_max1 = []
    temp_max2 = []
    for i in range(len(sizes)):
        temp.append([])
        for j in range(len(sizes[0])):
            temp[i].append(sizes[i][j])
        if temp[i][0] > temp[i][1]:
            temp[i][0],temp[i][1] = temp[i][1], temp[i][0]
    for i in range(len(sizes)):
        temp_max1.append(temp[i][0])
    for i in range(len(sizes)):
        temp_max2.append(temp[i][1])
    answer = max(temp_max1)*max(temp_max2)
    return answer